test/framework/resources/k8s/manifest/deployment.go

```diff
@@ -86,8 +89,34 @@ func (d *DeploymentBuilder) PodLabel(labelKey string, labelValue string) *Deploy
 	return d
 }
 
+func (d *DeploymentBuilder) HostNetwork(hostNetwork bool) *DeploymentBuilder {
+	d.hostNetwork = hostNetwork
+	return d
+}
+
+func (d *DeploymentBuilder) MountVolume(name string, mountpath string) *DeploymentBuilder {
```

Comment:
Oh are you saying that parameters should be of type Volume and VolumeMount instead of string type ? I can changed that So that we just do d.Volume = volume and d.VolumeMount = volumeMount ?

Reply:
On second thought what you have is fine as well,

Alternative could be (in case we want to add different types of volumes or multiple volumes in future),
```
func (d *DeploymentBuilder) Volume(name string, source corev1.VolumeSource) *DeploymentBuilder {
	d.volumes = append(d.volumes, corev1.Volume{
		Name:         name,
		VolumeSource: source,
	})
	return d
}
```
And in build,
```
Spec: corev1.PodSpec{
					Containers:                    []corev1.Container{d.container},
					TerminationGracePeriodSeconds: aws.Int64(int64(d.terminationGracePeriod)),
					NodeName:                      d.nodeName,
					Volumes:                       d.volumes,
				},
```